joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/MethodTests.scala

```diff
@@ -152,8 +152,74 @@ class MethodTests extends RubyCode2CpgFixture {
 
             case _ => fail("Expected two if statements for two default parameters")
           }
-        case _ => println("failed boss")
+        case _ => fail("Expected one method")
       }
     }
   }
+
+  "`def self.f(x) ... end` is represented by a METHOD inside the TYPE_DECL node" in {
+    val cpg = code("""
+                     |class C
+                     | def self.f(x)
+                     |  x + 1
+                     | end
+                     |end
+                     |""".stripMargin)
+
+    inside(cpg.typeDecl.name("C").l) {
+      case classC :: Nil =>
+        inside(classC.method.name("f").l) {
+          case funcF :: Nil =>
+            inside(funcF.parameter.l) {
+              case thisParam :: xParam :: Nil =>
+                thisParam.code shouldBe "this"
+                thisParam.typeFullName shouldBe "Test0.rb:<global>::program.C:f"
```

Comment:
The `this` parameter type should be the class, so only `Test0.rb:<global>::program.C`